src/main/scala/com/pingcap/tispark/TiUtils.scala

```diff
@@ -80,7 +80,10 @@ object TiUtils {
         // bit/duration type is not allowed to be pushed down
         case attr: AttributeReference if nameTypeMap.contains(attr.name) =>
           val head = nameTypeMap.get(attr.name).head
-          return !head.isInstanceOf[BitType] && head.getTypeCode != Types.TYPE_DURATION
+          return !head.isInstanceOf[BitType] &&
+            head.getTypeCode != Types.TYPE_DURATION &&
+            head.getTypeCode != Types.TYPE_TIMESTAMP &&
```

Comment:
This means TS and Datetime will not be pushed? Why is that?

Reply:
because afaik, spark does not push down timestamp and datetime in calculation.